app/src/main/java/org/mifos/selfserviceapp/ui/activities/HomeActivity.java

```diff
@@ -217,4 +228,76 @@ public void run() {
             super.onBackPressed();
         }
     }
+
+    private void checkPermissions() {
+        if (ActivityCompat.checkSelfPermission(this, Manifest.permission.CAMERA) !=
+                PackageManager.PERMISSION_GRANTED) {
+            ActivityCompat.requestPermissions(this, new String[]{Manifest.permission.CAMERA},
+                    CAMERA_PERMISSION);
+        }
+    }
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions,
+                                           @NonNull int[] grantResults) {
```

Comment:
@dilpreet96  Please look this https://github.com/openMF/android-client/blob/master/mifosng-android/src/main/java/com/mifos/utils/CheckSelfPermissionAndRequest.java